### DISCUSSION TITLE
Missing Thor class for invoke

### DISCUSSION THREAD

**Discussion by mubeguku** [CMDBrew/adminterface#112] [Q&A] [answered]
Any idea why I'm getting the following error on Mac after running "$ rails generate adminterface:install":

1: from /Users/xx/.rvm/gems/ruby-2.7.2/gems/adminterface-0.2.0/lib/generators/adminterface/install/install_generator.rb:8:in `install_rails_addons'
/Users/xx/.rvm/gems/ruby-2.7.2/gems/thor-1.1.0/lib/thor/invocation.rb:112:in `invoke': Missing Thor class for invoke action_text:install (RuntimeError)

**Comment by ilunglee** [comment, marked as answer]
Hello @mubeguku,
Thank you for reporting this issue. It seems to be a bug related to how the `install_generator` invokes the generator for `action_text`. I have created issue #113 and we will prepare a patch for this bug.

In the meanwhile, you can use the following to work around this issue:
```bash
# install active_admin
$ rails g active_admin:install --use_webpacker

# install action_text
$ bin/rails action_text:install

# install webpacker
$ bin/rails webpacker:install

# install adminterface assets using webpacker
$ rails g adminterface:webpacker

# install adminterface comments with action text
$ rails g adminterface:comments

# run the migration
$ rails db:migrate

# start the server
$ rails server
```

Thanks again for reporting the issue, and please let me know if the above works for you. 🙏

**Reply by ilunglee** [reply]
Hello @mubeguku,
I've fixed this issue on the `main` branch, and we will prepare a release in the next couple of days.

In the meantime, you can change your Gemfile to pick up the latest from Github.
```ruby
# Gemfile
gem "adminterface", git: "git@github.com:CMDBrew/adminterface.git", branch: "main"
```
And, now, you should be able to run the following without any issues.
```bash
$ bundle install
$ rails g adminterface:install
```

Thanks!

**Reply by ilunglee** [reply]
Fixed in [v0.2.1](https://github.com/CMDBrew/adminterface/releases/tag/v0.2.1):
Please update your Gemfile:
```ruby
# Gemfile
gem "adminterface", "0.2.1"
```
And, run:
```bash
# install gem
$ bundle install

# update npm package
$ yarn upgrade @cmdbrew/adminterface@0.2.1
```

Cheers,

**Comment by mubeguku** [comment]
Thanks a lot!!

Now I'm running into another issue on $ rails db:migrate

`-- rename_column(:active_admin_comments, :body, :body_backup)
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::UndefinedTable: ERROR:  relation "active_admin_comments" does not exist`

**Reply by ilunglee** [reply]
No worries!

Could you please check to see if `active_admin_comments` is in your schema (`db/schema.rb`)
```ruby
create_table "active_admin_comments" ...
```
and there is a migration file (i.e., `db/migrate/20210907171418_create_active_admin_comments.rb`— the `20210907171418` will be different based on the time you generated the file) for it?

If you don't see the above, likely, you have installed ActiveaAdmin without comments using the `--skip_comments` option:
```rails
rails g active_admin:install --skip_comments
```
If this is the case, you can safely remove the `drop_body_for_active_admin_comments.rb` migration file.

If you do see the migration file and the table inside the schema, please ensure the `drop_body_for_active_admin_comments.rb` has a timestamp prefix after the `create_active_admin_comments.rb`. For example,
- `db/migrate/20210907171418_create_active_admin_comments.rb`
- `db/migrate/20210907171454_drop_body_for_active_admin_comments.rb`

Let me know if this works for you!

Cheers,

**Reply by ilunglee** [reply]
@mubeguku,
I am following up to see if you have successfully resolved the issues.

Let me know. Thanks!

Best,

**Reply by mubeguku** [reply]
@ilunglee Sorry! Your solution worked. Thank you!

**Reply by ilunglee** [reply]
Awesome! Thanks for letting me know, @mubeguku.
